server/channels/app/limits.go

```diff
@@ -17,36 +15,37 @@ const (
 )
 
 func (a *App) GetServerLimits() (*model.ServerLimits, *model.AppError) {
-	var limits = &model.ServerLimits{}
-
-	if a.shouldShowUserLimits() {
-		activeUserCount, appErr := a.Srv().Store().User().Count(model.UserCountOptions{})
-		if appErr != nil {
-			mlog.Error("Failed to get active user count from database", mlog.String("error", appErr.Error()))
-			return nil, model.NewAppError("GetServerLimits", "app.limits.get_app_limits.user_count.store_error", nil, "", http.StatusInternalServerError).Wrap(appErr)
-		}
+	limits := &model.ServerLimits{}
+	license := a.License()
 
-		limits.ActiveUserCount = activeUserCount
+	if license == nil && maxUsersLimit > 0 {
+		// Enforce hard-coded limits for unlicensed servers.
 		limits.MaxUsersLimit = maxUsersLimit
 		limits.MaxUsersHardLimit = maxUsersHardLimit
+	} else if license.IsSeatCountEnforced && license.Features != nil && license.Features.Users != nil {
```

Comment:
We probably want to add a license nil check here as it could still be the case if `maxUserLimit ==0`